formats/gltf/writer.go

```diff
@@ -493,24 +493,46 @@ func (w *Writer) AddMesh(model PolyformModel) (_ int, err error) {
 	return meshIndex, nil
 }
 
-func (w *Writer) AddTexture(mat PolyformTexture) *TextureInfo {
-	newTex := &TextureInfo{Index: len(w.textures)}
+func (w *Writer) AddTexture(polyTex PolyformTexture) *TextureInfo {
+	newTexInfo := &TextureInfo{Index: len(w.textures)}
 
-	w.textures = append(w.textures, Texture{
+	newTex := Texture{
 		Sampler: ptrI(len(w.samplers)),
 		Source:  ptrI(len(w.images)),
-	})
+	}
+
+	texInfoExt := make(map[string]any)
+	texExt := make(map[string]any)
+	for _, ext := range polyTex.Extensions {
+		id := ext.TextureExtensionID()
+		if ext.IsInfo() {
+			texInfoExt[id] = ext.ToExtensionData(w)
+		} else {
+			texExt[id] = ext.ToExtensionData(w)
+		}
+
+		w.extensionsUsed[id] = true
```

Comment:
We might introduce a "required" method to the Texture Extension interface, and add a `Required` field to the `PolyformTextureTransform` that the user can set, and if it is, also add to `w.extensionsRequired`

https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_texture_transform
> *Implementation Note*: For maximum compatibility, it is recommended that exporters generate UV coordinate sets both with and without transforms applied, use the post-transform set in the texture texCoord field, then the pre-transform set with this extension. This way, if the extension is not supported by the consuming engine, the model still renders correctly. Including both will increase the size of the model, so if including the fallback UV set is too burdensome, either add this extension to extensionsRequired or use the same texCoord value in both places.